GVFS/GVFS.Common/GitHubUpgrader.cs

```diff
@@ -523,7 +523,27 @@ public bool TryLoad(out string error)
                 this.UpgradeRing = RingType.NoConfig;
 
                 string ringConfig = null;
-                if (this.LocalConfig.TryGetConfig(GVFSConstants.LocalGVFSConfig.UpgradeRing, out ringConfig, out error))
+                string upgradeFeedUrl = null;
+
+                // 1. Try to read ring from upgrade.feedurl first.
+                if (this.LocalConfig.TryGetConfig(GVFSConstants.LocalGVFSConfig.UpgradeFeedUrl, out upgradeFeedUrl, out error) &&

```

Comment:
As noted in my other comment - I don't think we should fall back to the GitHubUpgrader if there is an error to read the `UpgradeFeedUrl`, or it is non-empty. We should only load the GitHubUpgrader if `UpgraderFeedUrl` has been read, and it is empty (not present).

Alternatively, if we do not want to mix up `GitHubUpgrader` parsing `NuGetUpgrader` specific config settings, maybe we could read the config outside of `GitHubUpgrader`, and construct a `GitHubUpgrader` with an explicit ring... Someone has to read the config, but this might allow us to separate concerns a little bit.